main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    arr = self
+    arr = to_a if arr.is_a?(Range)
+    count = 0
+    while count < arr.size
+      yield arr[count]
+      count += 1
+    end
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    count = 0
+    arr = self
+    while count < arr.size
+      yield arr[count], count
+      count += 1
+    end
+  end
+
+  def my_select
+    return enum_for unless block_given? # if !block_given? return enum_for
+
+    array_with_selection = []
+    my_each do |x|
+      array_with_selection.push(x) if yield(x)
+    end
+    array_with_selection
+  end
+
+  def my_all?(*args)
+    arr = self
+    if args[0]
+      arr.my_each do |x|
+        return false unless x == args[0]
+
+        return true
+      end
+    else
+      arr.my_each do |x|
+        return false unless yield(x)
+
+        return true
+      end
+    end
+  end
+
+  def my_any?(*args)
+    arr = self
+    if args[0]
+      arr.my_each do |x|
+        return true unless x == args[0]
+
+        return false
+      end
+    else
+      arr.my_each do |x|
+        return true unless yield(x)
+
+        return false
+      end
+    end
+  end
+
+  def my_none?
+    !my_any?
+  end
+
+  def my_count(*args)
+    count = 0
+    if args[0]
+      my_each do |x|
+        count += 1 if x == args[0]
+      end
+    elsif !block_given?
+      count = size
+    elsif !args[0]
+      my_each do |x|
+        count += 1 if yield x
+      end
+    end
+    count
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains no truthy value.

- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method it's returning the elements that match the block passed instead of true if none meet the condition or false if at least one does.

- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains no truthy value.

- Your method doesn't consider the possibility when a class is given as an argument `(some_enumerable.my_none?(String)).`

- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_none?(/w/)).`

- Your method doesn't consider the possibility when a number is given as an argument `(some_enumerable.my_none?(8)).`


- Your method should work with ranges, too.

- Your method should work when you pass a block as an argument.